src/Lean/Elab/LetRec.lean

```diff
@@ -46,11 +46,12 @@ private def mkLetRecDeclView (letRec : Syntax) : TermElabM LetRecView := do
       addAuxDeclarationRanges declName decl declId
       let binders := decl[1].getArgs
       let typeStx := expandOptType declId decl[2]
-      let (type, numParams) ← elabBinders binders fun xs => do
+      let (type, binderIds) ← elabBindersEx binders fun xs => do
           let type ← elabType typeStx
           registerCustomErrorIfMVar type typeStx "failed to infer 'let rec' declaration type"
+          let (binderIds, xs) := xs.unzip
```

Comment:
Even though we use `unzip` in both callers of `elabBindersEx`, I still prefer an array of pairs over two separate arrays in the signature for documentation purposes, and assume that any potential overhead from this is not a concern here.

Reply:
An array of pairs is fine. I agree that the overhead is not a concern here.